src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -138,7 +138,33 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             else if (type == typeof(CodeBlockNodeModel))
             {
                 var code = obj["Code"].Value<string>();
-                node = new CodeBlockNodeModel(code, guid, 0.0, 0.0, libraryServices, ElementResolver);
+                CodeBlockNodeModel codeBlockNode = new CodeBlockNodeModel(code, guid, 0.0, 0.0, libraryServices, ElementResolver);
+                node = codeBlockNode;
+
+                // If the code block node is in an error state read the extra port data
+                // and initialize the input and output ports
+                if (node.IsInErrorState)
+                {
+                    List<string> inPortNames = new List<string>();
+                    var inputs = obj["Inputs"];
```

Comment:
can we pull out the string keys as constants? I think some of these might have already been pulled out by @QilongTang in a past PR.

Reply:
I can take a look into if "Inputs" has been saved as a constant elsewhere and see if we can use that instead. I agree that constants are better, but it will remain to be seen if they are accessible here in a way that makes using the constants better.

Reply:
After looking at the code it appears that there is not a constant for "Inputs" and the string is used directly wherever needed.